exporter/exporterhelper/batch_sender_test.go

```diff
@@ -535,6 +535,74 @@ func TestBatchSenderWithTimeout(t *testing.T) {
 	assert.EqualValues(t, 12, sink.itemsCount.Load())
 }
 
+func TestBatchSenderTimerResetNoConflict(t *testing.T) {
+	bCfg := exporterbatcher.NewDefaultConfig()
+	bCfg.MinSizeItems = 8
+	bCfg.FlushTimeout = 50 * time.Millisecond
+	be, err := newBaseExporter(defaultSettings, defaultDataType, newNoopObsrepSender,
+		WithBatcher(bCfg, WithRequestBatchFuncs(fakeBatchMergeFunc, fakeBatchMergeSplitFunc)))
+	require.NoError(t, err)
+	require.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))
+
+	sink := newFakeRequestSink()
+
+	// Send 2 concurrent requests that should be merged in one batch in the same interval as the flush timer
+	go func() {
+		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
+	}()
+	time.Sleep(50 * time.Millisecond)
+	go func() {
+		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
+	}()
+
+	// The batch should be sent either with the flush interval or by reaching the minimum items size with no conflict
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.LessOrEqual(c, uint64(1), sink.requestsCount.Load())
+		assert.EqualValues(c, 8, sink.itemsCount.Load())
+	}, 200*time.Millisecond, 10*time.Millisecond)
+
+	require.NoError(t, be.Shutdown(context.Background()))
+}
+
+func TestBatchSenderTimerFlush(t *testing.T) {
+	bCfg := exporterbatcher.NewDefaultConfig()
+	bCfg.MinSizeItems = 8
+	bCfg.FlushTimeout = 100 * time.Millisecond
+	be, err := newBaseExporter(defaultSettings, defaultDataType, newNoopObsrepSender,
+		WithBatcher(bCfg, WithRequestBatchFuncs(fakeBatchMergeFunc, fakeBatchMergeSplitFunc)))
+	require.NoError(t, err)
+	require.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))
+
+	sink := newFakeRequestSink()
+
+	time.Sleep(50 * time.Millisecond)
+
+	// Send 2 concurrent requests that should be merged in one batch and sent immediately
+	go func() {
+		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
+	}()
+	go func() {
+		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
+	}()
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.LessOrEqual(c, uint64(1), sink.requestsCount.Load())
+		assert.EqualValues(c, 8, sink.itemsCount.Load())
+	}, 30*time.Millisecond, 5*time.Millisecond)
+
+	// Send another request that should be sent after the flush timeout which supposed to happen after remaining 50ms
+	go func() {
+		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
+	}()
+
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		assert.LessOrEqual(c, uint64(1), sink.requestsCount.Load())
+		assert.EqualValues(c, 8, sink.itemsCount.Load())
+	}, 60*time.Millisecond, 5*time.Millisecond)
+
+	require.NoError(t, be.Shutdown(context.Background()))
+}
+
```

Comment:
- Update 1st test to make sure it deadlocks without the bugfix.
- Update 2nd test as comment "Send another request that should be sent after the flush timeout which supposed to happen after remaining 50ms" was incorrect imo. It should NOT flush in 50ms (the original ongoing timer), and should flush in 100ms (updated timer).

```suggestion
func TestBatchSenderTimerResetNoConflict(t *testing.T) {
	delayBatchMergeFunc := func(_ context.Context, r1 Request, r2 Request) (Request, error) {
		time.Sleep(30 * time.Millisecond)
		if r1 == nil {
			return r2, nil
		}
		fr1 := r1.(*fakeRequest)
		fr2 := r2.(*fakeRequest)
		if fr2.mergeErr != nil {
			return nil, fr2.mergeErr
		}
		return &fakeRequest{
			items:     fr1.items + fr2.items,
			sink:      fr1.sink,
			exportErr: fr2.exportErr,
			delay:     fr1.delay + fr2.delay,
		}, nil
	}

	bCfg := exporterbatcher.NewDefaultConfig()
	bCfg.MinSizeItems = 8
	bCfg.FlushTimeout = 50 * time.Millisecond
	be, err := newBaseExporter(defaultSettings, defaultDataType, newNoopObsrepSender,
		WithBatcher(bCfg, WithRequestBatchFuncs(delayBatchMergeFunc, fakeBatchMergeSplitFunc)))
	require.NoError(t, err)
	require.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))

	sink := newFakeRequestSink()

	// Send 2 concurrent requests that should be merged in one batch in the same interval as the flush timer
	go func() {
		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
	}()
	time.Sleep(30 * time.Millisecond)
	go func() {
		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
	}()

	// The batch should be sent either with the flush interval or by reaching the minimum items size with no conflict
	assert.EventuallyWithT(t, func(c *assert.CollectT) {
		assert.LessOrEqual(c, uint64(1), sink.requestsCount.Load())
		assert.EqualValues(c, 8, sink.itemsCount.Load())
	}, 200*time.Millisecond, 10*time.Millisecond)

	require.NoError(t, be.Shutdown(context.Background()))
}

func TestBatchSenderTimerFlush(t *testing.T) {
	bCfg := exporterbatcher.NewDefaultConfig()
	bCfg.MinSizeItems = 8
	bCfg.FlushTimeout = 100 * time.Millisecond
	be, err := newBaseExporter(defaultSettings, defaultDataType, newNoopObsrepSender,
		WithBatcher(bCfg, WithRequestBatchFuncs(fakeBatchMergeFunc, fakeBatchMergeSplitFunc)))
	require.NoError(t, err)
	require.NoError(t, be.Start(context.Background(), componenttest.NewNopHost()))

	sink := newFakeRequestSink()

	time.Sleep(50 * time.Millisecond)

	// Send 2 concurrent requests that should be merged in one batch and sent immediately
	go func() {
		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
	}()
	go func() {
		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
	}()

	assert.EventuallyWithT(t, func(c *assert.CollectT) {
		assert.LessOrEqual(c, uint64(1), sink.requestsCount.Load())
		assert.EqualValues(c, 8, sink.itemsCount.Load())
	}, 30*time.Millisecond, 5*time.Millisecond)

	// Send another request that should be flushed after 100ms instead of 50ms since last flush
	go func() {
		require.NoError(t, be.send(context.Background(), &fakeRequest{items: 4, sink: sink}))
	}()

	// Confirm that it is not flushed in 50ms
	time.Sleep(60 * time.Millisecond)
	assert.LessOrEqual(t, uint64(1), sink.requestsCount.Load())
	assert.EqualValues(t, 8, sink.itemsCount.Load())

	// Confirm that it is flushed after 100ms (using 60+50=110 here to be safe)
	time.Sleep(50 * time.Millisecond)
	assert.LessOrEqual(t, uint64(2), sink.requestsCount.Load())
	assert.EqualValues(t, 12, sink.itemsCount.Load())

	require.NoError(t, be.Shutdown(context.Background()))
}

```